synapse/handlers/profile.py

```diff
@@ -183,6 +191,12 @@ def set_avatar_url(self, target_user, requester, new_avatar_url, by_admin=False)
             target_user.localpart, new_avatar_url
         )
 
+        if self.hs.config.user_directory_include_pattern:
+            profile = yield self.store.get_profileinfo(target_user.localpart)
+            yield self.user_directory_handler.handle_local_profile_change(
+                target_user.user_id, profile
+            )
```

Comment:
I would move the if and fetching of profile into `handle_local_profile_change` rather than duplicating it everywhere

Reply:
mm, point.